File: dsa/linkedlist.py
class node:
    def __init__(self,data):
        self.data=data
        self.next=None

class linkedlist:
    def __init__(self):
        self.start=None

    def insert(self):
        d=int(input("value"))                  
        newnode=node(d)
        newnode.next=self.start
        self.start=newnode

    def insertend(self):
        if self.start is None:
            print("list is empty inserting the element to starting")
            d=int(input("value"))                  
            newnode=node(d)
            newnode.next=self.start
            self.start=newnode
        else:
            p=self.start
            while p.next is not None:
                p=p.next
            n=int(input("value"))
            newnode=node(n)
            newnode.next=None
            p.next=newnode


    def delete(self):
        if self.start is None:
            print ("list is empty")
        else:
            print(self.start.data)
            self.start=self.start.next

    def display(self):
        if self.start is None:
            print ("list is empty")
        else:
            temp=self.start
            while temp is not None:
                print(temp.data)
                temp=temp.next

if __name__=="__main__":
    l1=linkedlist()

    while(1):

        n=int(input("choice"))


        if (n == 1):
            l1.insert()

        elif (n ==5):
            l1.insertend()

        elif (n == 2):
            l1.delete()
        elif (n == 3):
            l1.display()
        elif (n == 4):
            print("exiting")
            break
        else:
            print("wrong choice")

File: c program/linkedlist.c
#include <stdio.h>
#include <stdlib.h>

struct node
{
    int data;
    struct node *next;
};

void insert(struct node **start);
void insertend(struct node **start);
void insertspe(struct node **start);
 void delete(struct node **start);
void display(struct node *start);

int main()
{
    int value;
    struct node *start = NULL;

    while (1)
    {
        printf("enter your choice");
        scanf("%d", &value);

        switch (value)
        {
        case 1:
            insert(&start);

            break;

        case 2:
            insertend(&start);
            break;

        case 3:
            insertspe(&start);
            break;

        case 4:
            delete (&start);
            break;

        case 5:
            display(start);
            break;

        case 6:
            printf("exiting...");
            return 0;

        default:
            printf("wrong choice"
                   "\n");
            break;
        }
    }
}

void insert(struct node **start)
{
    int v;
    printf("enter the value ");
    scanf("%d", &v);
    struct node *newnode = (struct node *)malloc(sizeof(struct node));

    newnode->data = v;
    newnode->next = *start;
    *start = newnode;

    printf("inserted");
    printf("\n");
}

void insertend(struct node **start)
{
    int e;
    printf("enter the value ");
    scanf("%d", &e);

    struct node *newnode = (struct node *)malloc(sizeof(struct node));

    struct node *p = *start;
    if (*start == NULL)
    {
        printf("list is empty");
        printf("\n");
        printf("data is inserted to the begining");

        newnode->data = e;
        newnode->next = *start;
        *start = newnode;
    }
    else
    {
        while (p->next != NULL)
        {
            p = p->next;
        }
        newnode->data = e;
        newnode->next = NULL;
        p->next = newnode;
    }
    printf("\n");
}

void insertspe(struct node **start)
{
    int s;
    printf("enter the data");
    scanf("%d", &s);

    struct node *newnode = (struct node *)malloc(sizeof(struct node));

    if (*start == NULL)
    {
        printf("list is empty");
        printf("\n");
        printf("data is inserted to the begining");

        newnode->data = s;
        newnode->next = *start;
        *start = newnode;
    }
    else
    {
        int l;
        printf("enter the loc");
        scanf("%d", &l);
        struct node *p = *start;

        while (p!=NULL&&p->data != l)
        {
            p = p->next;
        }
        newnode->data = s;
        newnode->next = p->next;
        p->next = newnode;
    }
}

void delete(struct node **start)
{
    if (*start == NULL)
    {
        printf("list is empty");
        return;
    }

    struct node *p = *start;
    *start = (*start)->next;
    printf("%d", p->data);
    free(p);
    printf("\n");
}

void display(struct node *start)
{
    if (start == NULL)
    {
        printf("list is empty");
        return;
    }

    struct node *p = start;
    while (p != NULL)

    {
        printf("%d->", p->data);
        p = p->next;
    }
    printf("NULL"
           "\n");
}